consul/assets/configuration/spec.yaml

```diff
@@ -38,6 +38,14 @@ files:
       value:
         type: boolean
         example: false
+    - name: use_node_id_as_hostname
+      description: |
+        Whether to use the Consul node name as the host name for network latency metrics.
+        If set to true, the agent will use the Consul node name as the host name.
+        If set to false, the agent will use the system hostname as the host name.
```

Comment:
Non blocking nit. I think the description could maybe describe the behavior a bit more? Like what happens if the agent uses the consul node name as the host name? I think the PR description has a good enough explanation.